### PR TITLE
Invalidate old tokens after reset password

### PR DESCRIPTION
**Pull request by zkhin**
Fixes: https://github.com/SeedCompany/cord-api-v3/issues/1645

### CODE REVIEW

File: src/components/authentication/authentication.service.ts [hunk 1]
```diff
@@ -404,12 +404,24 @@ export class AuthenticationService {
...
+      .query()
+      .match([
+        node('emailAddress', 'EmailAddress', { value: result.email }),
+        relation('in', '', 'email', { active: true }),
+        node('user', 'User'),
+        relation('out', 'oldRel', 'token', { active: true }),
+        node('token', 'Token'),
+      ])
+      .setValues({ 'oldRel.active': false })
+      .run();
```
Comment (CarsonF) [inline src/components/authentication/authentication.service.ts:424]: We still need to avoid the current session here.

**Review by CarsonF — CHANGES_REQUESTED** [review]

File: src/components/authentication/authentication.service.ts (hunk 1 above)
Comment (zkhin) [reply src/components/authentication/authentication.service.ts:424]: Do we have a valid `session.token` while resetting the password?

Comment (zkhin) [reply src/components/authentication/authentication.service.ts:424]: I don't think we have `LoggedInSession` in the `resetPassword` resolver.

Comment (CarsonF) [reply src/components/authentication/authentication.service.ts:424]: Ah. You can add an anonymous session arg, which has a token, just `userId` is optional.
```ts
@AnonSession() session: Session
```

Comment (zkhin) [reply src/components/authentication/authentication.service.ts:428]: Done

**Review by CarsonF — APPROVED** [review]
Sweet!

**Pull request merged**